lib/rspec/json_expectations/json_traverser.rb

```diff
@@ -53,13 +53,31 @@ def handle_array(errors, expected, actual, negate=false, prefix=[])
           handle_keyvalue(errors, transformed_expected, actual, negate, prefix)
         end
 
-        def handle_unordered(errors, expected, actual, negate=false, prefix=[])
+        def handle_unordered(errors, expected, actual, negate=false, prefix=[], options={})
           return nil unless expected.is_a?(Matchers::UnorderedArrayMatcher)
 
-          conditionally_negate(
-            expected.match(errors, actual, prefix),
-            negate
-          )
+          match_size_assertion_result = \
+            match_size_of_collection(errors, expected, actual, prefix, options)
+
+          if match_size_assertion_result == false
+            false
```

Comment:
Feels like there should be `conditionally_negate(false, negate)` here. Also, that means, that we probably need additional feature test for this.